lib/client/tcp-client.js

```diff
@@ -30,27 +30,33 @@ TcpClient.prototype.connect = function(callback) {
       self.responseBuffer += data.toString();
       if (self.responseBuffer.substring(self.responseBuffer.length - 2, self.responseBuffer.length) == FS + CR) {
         var ack = self.parser.parse(self.responseBuffer.substring(1, self.responseBuffer.length - 2));
-        self.callback(null, ack);
         self.responseBuffer = "";
         self.awaitingResponse = false;
         if (!self.keepalive) {
           self.close();
         }
+        self.callback(null, ack);
```

Comment:
wouldn't allow another message to be queued up right after the first one completed